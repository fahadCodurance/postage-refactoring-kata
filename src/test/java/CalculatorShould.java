import org.junit.Before;
import org.junit.Test;

import static org.approvaltests.combinations.CombinationApprovals.verifyAllCombinations;

public class CalculatorShould {

    private Calculator calculator;

    @Before
    public void setUp() {
        calculator = new Calculator();
    }

    @Test
    public void calculate_cost() {
        verifyAllCombinations(
                this::doCalculation,
                new Integer[]{0, 60, 61, 500, 502},
                new Integer[]{0, 229, 230, 324, 325},
                new Integer[]{0, 162, 163, 229, 230},
                new Integer[]{0, 25, 26, 100, 101},
                new Currency[]{Currency.Gbp, Currency.Chf, Currency.Eur, null}
        );
    }

    private String doCalculation(int weight, int height, int width, int depth, Currency currency) {
        return calculator.calculate(weight, height, width, depth, currency).toString();
    }

    @Test
    public void calculate_cost_with_package() {
        verifyAllCombinations(
                this::doCalculationWithPackage,
                new Package[]{
                        new Package(0, 0, 0, 0),
                        new Package(60, 229, 162, 25),
                        new Package(61, 230, 163, 26),
                        new Package(500, 324, 229, 100),
                        new Package(502, 325, 230, 101)},
                new Currency[]{Currency.Gbp, Currency.Chf, Currency.Eur, null}
        );

    }

    private String doCalculationWithPackage(Package aPackage, Currency currency) {
        return calculator.calculate(aPackage, currency).toString();
    }
}
