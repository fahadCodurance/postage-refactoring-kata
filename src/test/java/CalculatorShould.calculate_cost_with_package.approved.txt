[Package{weight=0, height=0, width=0, depth=0}, Gbp] => Money{currency=Gbp, amount=120.00} 
[Package{weight=0, height=0, width=0, depth=0}, Chf] => Money{currency=Chf, amount=435.20} 
[Package{weight=0, height=0, width=0, depth=0}, Eur] => Money{currency=Eur, amount=400.00} 
[Package{weight=0, height=0, width=0, depth=0}, null] => Cannot invoke "Money.toString()" because the return value of "Calculator.calculate(Package, Currency)" is null 
[Package{weight=60, height=229, width=162, depth=25}, Gbp] => Money{currency=Gbp, amount=120.00} 
[Package{weight=60, height=229, width=162, depth=25}, Chf] => Money{currency=Chf, amount=435.20} 
[Package{weight=60, height=229, width=162, depth=25}, Eur] => Money{currency=Eur, amount=400.00} 
[Package{weight=60, height=229, width=162, depth=25}, null] => Cannot invoke "Money.toString()" because the return value of "Calculator.calculate(Package, Currency)" is null 
[Package{weight=61, height=230, width=163, depth=26}, Gbp] => Money{currency=Gbp, amount=244.00} 
[Package{weight=61, height=230, width=163, depth=26}, Chf] => Money{currency=Chf, amount=603.84} 
[Package{weight=61, height=230, width=163, depth=26}, Eur] => Money{currency=Eur, amount=555.00} 
[Package{weight=61, height=230, width=163, depth=26}, null] => Cannot invoke "Money.toString()" because the return value of "Calculator.calculate(Package, Currency)" is null 
[Package{weight=500, height=324, width=229, depth=100}, Gbp] => Money{currency=Gbp, amount=2000.00} 
[Package{weight=500, height=324, width=229, depth=100}, Chf] => Money{currency=Chf, amount=2992.00} 
[Package{weight=500, height=324, width=229, depth=100}, Eur] => Money{currency=Eur, amount=2750.00} 
[Package{weight=500, height=324, width=229, depth=100}, null] => Cannot invoke "Money.toString()" because the return value of "Calculator.calculate(Package, Currency)" is null 
[Package{weight=502, height=325, width=230, depth=101}, Gbp] => Money{currency=Gbp, amount=45294.00} 
[Package{weight=502, height=325, width=230, depth=101}, Chf] => Money{currency=Chf, amount=61871.84} 
[Package{weight=502, height=325, width=230, depth=101}, Eur] => Money{currency=Eur, amount=56867.50} 
[Package{weight=502, height=325, width=230, depth=101}, null] => Cannot invoke "Money.toString()" because the return value of "Calculator.calculate(Package, Currency)" is null 
